woocommerce/woocommerce-gutenberg-products-block
# Add `CopyWebpackPlugin` to ensure `block.json` is copied to `build`

This PR adds the `CopyWebpackPlugin` plugin to copy the `block.json` file from `assets/js/blocks/checkout` to `build/checkout/block.json`. 

The `get_block_metadata_path` has also been updated to get the metadata from `build` instead of assets.

In #5594 changes were made to register the block from this metadata file, in built contexts, the metadata file was not present.

At the moment the `from` and `to` values are hard-coded in the interests of getting the release out, but we will follow up in a later PR to make this dynamic.

### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

1. Run `npm run start` .
2. Modify `assets/js/blocks/checkout/block.json`.
3. Open `build/checkout/block.json` and ensure the contents reflect your edit from step 2.
4. Revert your changes, stop the `start` script.
2. Run `npm run build` - check the `build/checkout/block.json` file is present.
3. Run `npm run package-plugin:deploy`.
4. Upload the testing zip to a test site.
5. Ensure you can add the checkout block to a page.
6. View it on the front end and ensure it works, make a test order.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

1. Go to a page and try to add the checkout block.
2. Make a test order using the checkout block.

### Changelog

> Skipping
